weatherman.py

```diff
@@ -0,0 +1,246 @@
+
+    @staticmethod
+    def generate_barchart(temp, is_highest):
+        if is_highest:
+            return "\033[91m {}\033[00m".format("+" * temp)
+        return "\033[94m {}\033[00m".format("+" * temp)
+
+    @staticmethod
+    def maxmin_temperature_on_days(year, month):
+        for day in __weather__["Murree"][year][month]:
+            temp = Computations.maximum_temperature_on_day(year, month, day)
+            temp_ = Computations.minimum_temperature_on_day(year, month, day)
+            if temp:
+                print(str(day) + " " + Report.generate_barchart(temp, True) + " " + str(temp) + "C")
+            if temp_:
+                print(str(day) + " " + Report.generate_barchart(temp_, False) + " " + str(temp_) + "C")
+
+    @staticmethod
+    def maxmin_temperature_on_days_bonus(year, month):
+        for day in __weather__["Murree"][year][month]:
+            temp = Computations.maximum_temperature_on_day(year, month, day)
+            temp_ = Computations.minimum_temperature_on_day(year, month, day)
+            if temp and temp_:
+                print(str(day) + " " + Report.generate_barchart(temp_, False) + Report.generate_barchart(temp, True)
+                      + " " + str(temp_) + "C" + " " + str(temp) + "C")
+            elif temp:
+                print(str(day) + " " + Report.generate_barchart(temp, True) + " " + str(temp) + "C")
+            elif temp_:
+                print(str(day) + " " + Report.generate_barchart(temp_, False) + " " + str(temp_) + "C")
+
+
+if __name__ == '__main__':
+    # Command Line Arguments
+    # first: path of weather-files, if weather-files are in the same directory of python file just type None
+    # second: operation (-e, -a, -c)
+    # third: argument : year, month etc
+    if len(sys.argv) < 4:
+        print("Less no of Arguments")
+    else:
+        parser = Parser()
+        if sys.argv[1] != "None":
+            print(sys.argv[1])
+            parser.read_folder(sys.argv[1])
+        else:
+            parser.read_folder('./weatherfiles')
+        arguments = len(sys.argv) - 2
+        index = 2
+        while arguments > 0:
+            if sys.argv[index] == "-e":
+                year_ = sys.argv[index+1]
+                Report.maximum_temperature(year_)
+                Report.minimum_temperature(year_)
+                Report.maximum_humidity(year_)
+            elif sys.argv[index] == "-a":
+                year_, month_ = sys.argv[index+1].split("/")
+                month_ = __months__[int(month_)]
+                Report.average_highest_temperature(year_, month_)
+                Report.average_lowest_temperature(year_, month_)
+                Report.average_mean_humidity(year_, month_)
+            elif sys.argv[index] == "-c":
+                year_, month_ = sys.argv[index + 1].split("/")
+                month_ = __months__[int(month_)]
+                Report.maxmin_temperature_on_days(year_, month_)
+                # for bonus task uncomment the below line
```

Comment:
If one uses comments in the code to explain something, it means code is not that good to be self explanatory. Read it twice!